### PR TITLE
Added search/search_tests

### PR DESCRIPTION
**Pull request by therearesomewhocallmetim**
search_integration_tests are to come as they depend on generator_tests_support.

### CODE REVIEW

File: search/search_tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  query_saver_tests.cpp
+  ranking_tests.cpp
+  string_intersection_test.cpp
+  string_match_test.cpp
+)
+
+omim_add_test(search_tests ${SRC})
+
+omim_link_libraries(
+  search_tests search
```
Comment (syershov) [inline search/search_tests/CMakeLists.txt]: На разны строки перенеси

File: 3party/protobuf/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    SRC
+    ${SRC}
+    src/google/protobuf/stubs/atomicops_internals_x86_gcc.cc
+  )
+endif()
+
+if (PLATFORM_WIN AND MSVC)
+  set(
+    SRC
+    src/google/protobuf/stubs/atomicops_internals_x86_msvc.cc
```
Comment (syershov) [inline 3party/protobuf/CMakeLists.txt]: Забыл ${SRC}

File: search/search_tests/CMakeLists.txt (hunk 1 above)
Comment (therearesomewhocallmetim) [inline search/search_tests/CMakeLists.txt]: Done

File: 3party/protobuf/CMakeLists.txt (hunk 2 above)
Comment (therearesomewhocallmetim) [inline 3party/protobuf/CMakeLists.txt]: Done

**Comment by therearesomewhocallmetim** [comment]
PTAL

File: indexer/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  tree_structure.hpp
+  trie_builder.hpp
+  trie_reader.hpp
+  trie.hpp
+  types_mapping.cpp
+  types_mapping.hpp
+  unique_index.hpp
+)
+
+
```
Comment (syershov) [inline indexer/CMakeLists.txt]: лишняя строчка

**Comment by therearesomewhocallmetim** [comment]
PTAL

File: search/search_integration_tests/CMakeLists.txt [hunk 4]
```diff
@@ -5,15 +5,29 @@ add_compile_options(
...
-#endif()
 
-# omim_add_test_subdirectory(search_integration_tests_tests)
-omim_add_test(search_integration_tests ${SRC})
-omim_link_libraries(search_integration_tests ...)
\ No newline at end of file
+)
+
+if (PLATFORM_MAC AND NOT NO_TESTS)
+  target_link_libraries(
```
Comment (ygorshenin) [inline search/search_integration_tests/CMakeLists.txt]: Can we use omim_link_libraries here?

File: 3party/protobuf/CMakeLists.txt [hunk 5]
```diff
@@ -66,6 +66,7 @@ endif()
 if (PLATFORM_WIN AND MSVC)
   set(
     SRC
+    ${SRC}
```
Comment (ygorshenin) [inline 3party/protobuf/CMakeLists.txt]: These set(A ${A} ...) looks ugly. Can we use cmake lists here instead? The expression above will look like:
```cmake
list(APPEND SRC src/google/...)
```

File: search/search_integration_tests/CMakeLists.txt (hunk 4 above)
Comment (therearesomewhocallmetim) [inline search/search_integration_tests/CMakeLists.txt]: Yes, when we do omim_integration_tests.

File: indexer/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  new_feature_categories.cpp   # it's in indexer because of CategoriesHolder dependency.
+  new_feature_categories.hpp   # it's in indexer because of CategoriesHolder dependency.
+  old/feature_loader_101.cpp
+  old/feature_loader_101.hpp
+  old/interval_index_101.hpp
+  osm_editor.cpp
+  osm_editor.hpp
+  point_to_int64.cpp
+  point_to_int64.hpp
+  postcodes_matcher.cpp   # it's in indexer due to editor wich is in indexer and depends on postcodes_marcher
```
Comment (ygorshenin) [inline indexer/CMakeLists.txt]: wich?

File: indexer/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+  postcodes_matcher.hpp    # it's in indexer due to editor wich is in indexer and depends on postcodes_marcher
```
Comment (ygorshenin) [inline indexer/CMakeLists.txt]: wich wich?

File: search/search_integration_tests/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  pre_ranker_test.cpp
+  processor_test.cpp
+  search_edited_features_test.cpp
+  smoke_test.cpp
+)
+
+omim_add_test(search_integration_tests ${SRC})
+omim_link_libraries(
+  search_integration_tests
+
```
Comment (ygorshenin) [inline search/search_integration_tests/CMakeLists.txt]: Sorry?

File: search/search_integration_tests/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+)
+
+if (PLATFORM_MAC AND NOT NO_TESTS)
+  target_link_libraries(
```
Comment (therearesomewhocallmetim) [inline search/search_integration_tests/CMakeLists.txt]: Removed the file, because I haven't implemented the integration tests yet.

File: 3party/protobuf/CMakeLists.txt (hunk 5 above)
Comment (therearesomewhocallmetim) [inline 3party/protobuf/CMakeLists.txt]: I agree that they look ugly, but I suggest replacing them in a separate PR, as it will affect many files that are not directly related to this pull-request.

File: indexer/CMakeLists.txt (hunk 7 above)
Comment (therearesomewhocallmetim) [inline indexer/CMakeLists.txt]: Done

**Comment by therearesomewhocallmetim** [comment]
PTAL

File: search/search_tests/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,58 @@
+include_directories(${OMIM_ROOT}/3party/jansson/src)
+
+add_compile_options(
```
Comment (ygorshenin) [inline search/search_tests/CMakeLists.txt]: Why do we need this? These options should come from the parent cmakelist, right?

**Comment by therearesomewhocallmetim** [comment]
PTAL

**Comment by therearesomewhocallmetim** [comment]
DONT MERGE JUST YET

**Comment by syershov** [comment]
LGTM

File: indexer/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,144 @@
+include_directories(${OMIM_ROOT}/3party/protobuf/src)
+
+add_compile_options(
```
Comment (ygorshenin) [inline indexer/CMakeLists.txt]: Why? These options should be set in the parent CMakeLists.

File: search/CMakeLists.txt [hunk 12]
```diff
@@ -0,0 +1,138 @@
+add_compile_options(
```
Comment (ygorshenin) [inline search/CMakeLists.txt]: Same there - these options must be set once, in the parent CMakeLists.

File: search/search_tests/CMakeLists.txt [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+  string_match_test.cpp
+)
+
+omim_add_test(search_tests ${SRC})
+
+omim_link_libraries(
+  search_tests
+  search
+  indexer
+  platform
```
Comment (ygorshenin) [inline search/search_tests/CMakeLists.txt]: platform should go after editor, but before coding.

**Comment by mapsmetest** [comment]
PTAL

**Comment by syershov** [comment]
LGTM

File: 3party/protobuf/CMakeLists.txt [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+
+include_directories(. src ../../)
+
+add_compile_options(
+  "-Wall"
+  "-std=c++11"
+)
+
+if (NOT PLATFORM_WIN)
+  add_definitions(-DHAVE_PTHREAD)
```
Comment (ygorshenin) [inline 3party/protobuf/CMakeLists.txt]: Maybe it worth to find_package(Threads)?

File: 3party/protobuf/CMakeLists.txt [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+
+if (PLATFORM_WIN AND MSVC)
+  set(
+    SRC
+    ${SRC}
+    src/google/protobuf/stubs/atomicops_internals_x86_msvc.cc
+  )
+endif()
+
+add_library(protobuf ${SRC})
```
Comment (ygorshenin) [inline 3party/protobuf/CMakeLists.txt]: Use project name variable.

File: editor/CMakeLists.txt [hunk 16]
```diff
@@ -1,3 +1,5 @@
+project(editor)
```
Comment (ygorshenin) [inline editor/CMakeLists.txt]: Am I right that compiler options will be derived here?

Comment (ygorshenin) [inline editor/CMakeLists.txt]: From the parent CMakeList?

File: search/search_tests_support/CMakeLists.txt [hunk 17]
```diff
@@ -0,0 +1,18 @@
+project(search_tests_support)
+
+add_compile_options(
+  "-Wall"
```
Comment (ygorshenin) [inline search/search_tests_support/CMakeLists.txt]: Not good, as this duplicates compiler options from the root CMakeLists.

File: tracking/tracking_tests/CMakeLists.txt [hunk 18]
```diff
@@ -1,14 +1,26 @@
...
 set(
   SRC
   protocol_test.cpp
   reporter_test.cpp
 )
 
-omim_add_test(tracking_tests ${SRC})
-omim_link_libraries(tracking_tests tracking platform platform_tests_support coding base geometry tomcrypt ${Qt5Widgets_LIBRARIES})
+omim_add_test(${PROJECT_NAME} ${SRC})
+omim_link_libraries(
```
Comment (ygorshenin) [inline tracking/tracking_tests/CMakeLists.txt]: Blank line, please.

File: 3party/protobuf/CMakeLists.txt (hunk 14 above)
Comment (mapsmetest) [inline 3party/protobuf/CMakeLists.txt]: In general, this file will not work independently of our main project because we use the PLATFORM_WIN variable, so threads will have been found in the main project.

File: 3party/protobuf/CMakeLists.txt (hunk 15 above)
Comment (mapsmetest) [inline 3party/protobuf/CMakeLists.txt]: ok

File: editor/CMakeLists.txt (hunk 16 above)
Comment (mapsmetest) [inline editor/CMakeLists.txt]: Yes

**Comment by mapsmetest** [comment]
PTAL

**Comment by ygorshenin** [comment]
LGTM
